python/pyarrow/io.pxi

```diff
@@ -677,16 +676,19 @@ cdef class NativeFile(_Weakrefable):
 
         self.seek(0)
 
-        writer_thread = threading.Thread(target=bg_write)
+        if is_threading_enabled():
+            writer_thread = threading.Thread(target=bg_write)
+        else:
+            writer_thread = None
 
         # This isn't ideal -- PyBytes_FromStringAndSize copies the data from
         # the passed buffer, so it's hard for us to avoid doubling the memory
         buf = <uint8_t*> malloc(buffer_size)
         if buf == NULL:
             raise MemoryError("Failed to allocate {0} bytes"
                               .format(buffer_size))
-
-        writer_thread.start()
+        if writer_thread:
```

Comment:
Instead of sprinkling `if writer_thread` conditionals everywhere, I think this would be clearer if there were two separate implementations.